be/src/vec/exec/scan/scanner_context.h

```diff
@@ -272,6 +272,8 @@ class ScannerContext {
     // Not need to protect by lock, because only one scheduler thread will access to it.
     std::mutex _scanners_lock;
     std::list<VScannerSPtr> _scanners;
+    // weak pointer for _scanners, used in stop function
+    std::vector<VScannerWPtr> _scanners_ref;
```

Comment:
There is already a     std::list<VScannerSPtr> _scanners;
why do we need a weak ptr here?

Reply:
Because `_scanners` will be popped/pushed/released when running, and `try_close` has no need for synchronization and consistency, only do the best effort, so there's no need to use `_scanners` with some lock operations.